app/marketing/tasks.py

```diff
@@ -1,20 +1,99 @@
 from django.conf import settings
 
+import csv
+import os
 import marketing.stats as stats
 from app.services import RedisService
 from celery import app
 from celery.utils.log import get_task_logger
+from django.contrib.auth.models import User
+from django.utils import timezone
 from marketing.mails import new_bounty_daily as new_bounty_daily_email
-from marketing.mails import weekly_roundup as weekly_roundup_email
+from marketing.mails import weekly_roundup as weekly_roundup_email, send_mail
 from marketing.models import EmailSubscriber
 from marketing.utils import allowed_to_send_email
+from retail.emails import render_export_data_email, render_export_data_email_failed
 
 logger = get_task_logger(__name__)
 
 redis = RedisService().redis
 
 rate_limit = '300000/s' if settings.FLUSH_QUEUE or settings.MARKETING_FLUSH_QUEUE else settings.MARKETING_QUEUE_RATE_LIMIT
 
+def create_csv(export_type, profile, earnings):
+    path = f'app/assets/tmp/user-{export_type}/{profile}'
+    if not os.path.isdir(path):
+        os.makedirs(path)
+    name = f"{timezone.now().strftime('%Y_%m_%dT%H')}"
+    file_path = f'{path}/{name}.csv'
```

Comment:
Should we be saving these to disk? Could we use something like `io.StringIO()` to keep these in memory?